jujugui/static/gui/src/app/components/error-boundary/test-error-boundary.js

```diff
@@ -0,0 +1,44 @@
+/* Copyright (C) 2018 Canonical Ltd. */
+'use strict';
+
+const React = require('react');
+const enzyme = require('enzyme');
+const ErrorBoundary = require('./error-boundary');
+
+class ComponentWithError extends React.Component {
+  constructor(props) {
+    super(props);
+    throw new Error('Error!');
+  }
+  render() {
+    return (
+      <div>
+        <h1>Buggy component</h1>
+      </div>
+    );
+  }
+}
+
+describe('Error boundary', function() {
+
+  const renderComponent = (options = {}) => enzyme.shallow(
+    <ErrorBoundary />
```

Comment:
These tests are no good because you're not actually passing the content into the component ;)

Reply:
I pass the content object in the next code block, line 30 ;) Why? This is a pattern Huw uses elsewhere which I replicated for consistency.

Reply:
@hatched Right.. gotcha. I've updated the tests but the snapshot for the bug component still doesn't throw the fall back ui? Can you see what I'm missing?

Reply:
Because the broken component is child component of the wrapper it's probably not being executed because of the shallow render. There are hacks to make it render but you should probably render it to the dom. 

See: https://github.com/juju/juju-gui/blob/use-jujulib/jujugui/static/gui/src/app/components/profile/credential-list/test-credential-list.js#L67

And then be sure to clean up afterwards: https://github.com/juju/juju-gui/blob/use-jujulib/jujugui/static/gui/src/app/components/profile/credential-list/test-credential-list.js#L107